sn_client/src/audit/dag_crawling.rs

```diff
@@ -400,47 +408,49 @@ impl Client {
                     }
                 }
                 let spends_len = spends.len();
-                debug!("Depth {depth} - Got {spends_len} spends for parent Tx: {parent_tx_hash:?}");
-                trace!("Spends for {parent_tx_hash:?} - {spends:?}");
-
-                // check if we reached the genesis Tx
-                known_txs.insert(parent_tx_hash);
-                if parent_tx == *sn_transfers::GENESIS_CASHNOTE_PARENT_TX
-                    && spends
-                        .iter()
-                        .all(|s| s.spend.unique_pubkey == *sn_transfers::GENESIS_SPEND_UNIQUE_KEY)
+                debug!("Depth {depth} - Got {spends_len} spends for parent: {addrs_to_verify:?}");
+                trace!("Spends for {addrs_to_verify:?} - {spends:?}");
+
+                // check if we reached the genesis spend
+                known_ancestors.extend(addrs_to_verify.clone());
+                if spends
+                    .iter()
+                    .all(|s| s.spend.unique_pubkey == *sn_transfers::GENESIS_SPEND_UNIQUE_KEY)
                     && spends.len() == 1
```

Comment:
Got to check if this check is still accurate, wondering if we should not just
- check for genesis spends (not just addr, but content too) and remove them
- keep going for the others